index.js

```diff
@@ -216,4 +216,16 @@ function _createXHR(options) {
 
 }
 
+function getXml(xhr) {
+    if (xhr.responseType === "document") {
+        return xhr.responseXML
+    }
+    var firefoxBugTakenEffect = xhr.status === 204 && xhr.responseXML && xhr.responseXML.documentElement.nodeName === "parsererror"
```

Comment:
This is too restrictive. 201, 202 and 205 are also affected.
cf https://bugzilla.mozilla.org/show_bug.cgi?id=884693#c34